Support Windows ARM Runners

GitHub added Windows Arm64 hosted runners:

https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

The [`windows-11-arm`](https://github.com/actions/partner-runner-images/blob/main/images/arm-windows-11-image.md) image unfortunately comes without `rustup` installed, resulting in the following error:

https://github.com/maennchen/gleam/actions/runs/14616757309/job/41006654440

```
Run actions-rust-lang/setup-rust-toolchain@v1
  with:
    toolchain: stable
    target: aarch64-pc-windows-msvc
    cache-key: v1-aarch64-pc-windows-msvc
    cache: true
    cache-on-failure: true
    cache-bin: true
    matcher: true
    rustflags: -D warnings
    override: true
  env:
    CARGO_TERM_COLOR: always
    RUSTFLAGS: -D warnings
Run : construct rustup command line
  : construct rustup command line
  echo "targets=$(for t in ${targets//,/ }; do echo -n ' --target' $t; done)" >> $GITHUB_OUTPUT
  echo "components=$(for c in ${components//,/ }; do echo -n ' --component' $c; done)" >> $GITHUB_OUTPUT
  echo "downgrade=" >> $GITHUB_OUTPUT
  shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
  env:
    CARGO_TERM_COLOR: always
    RUSTFLAGS: -D warnings
    targets: aarch64-pc-windows-msvc
    components: 
  
Run if [[ ! -v CARGO_INCREMENTAL ]]; then
Run echo "::add-matcher::C:\a\_actions\actions-rust-lang\setup-rust-toolchain\v1/rust.json"
  
Run if [[ -z "$toolchain" && ( -f "rust-toolchain" || -f "rust-toolchain.toml" ) ]]
C:\a\_temp\6dbba988-4564-4c[39](https://github.com/maennchen/gleam/actions/runs/14616757309/job/41006654440#step:4:43)-89e8-552e03bf3b06.sh: line 19: rustup: command not found
Error: Process completed with exit code 127.
```

The action seems to skip install for Windows in general (same issue as #45):

https://github.com/actions-rust-lang/setup-rust-toolchain/blob/9399c7bb15d4c7d47b27263d024f0a4978346ba4/action.yml#L132-L139

It would be nice to support Windows rustup installation and therefore all GH hosted runners.

--- a/action.yml
+++ b/action.yml
@@ -130,13 +130,20 @@ runs:
       run: echo "::add-matcher::${{ github.action_path }}/rust.json"
 
     - name: Install rustup, if needed
-      if: runner.os != 'Windows'
       shell: bash
       run: |
         if ! command -v rustup &> /dev/null ; then
           curl --proto '=https' --tlsv1.2 --retry 10 --retry-connrefused -fsSL "https://sh.rustup.rs" | sh -s -- --default-toolchain none -y
-          echo "${CARGO_HOME:-$HOME/.cargo}/bin" >> $GITHUB_PATH
+          
+          # Resolve the correct CARGO_HOME path depending on OS
+          if [[ "$RUNNER_OS" == "Windows" ]]; then
+            echo "${CARGO_HOME:-$USERPROFILE/.cargo}/bin" | sed 's|/|\\|g' >> $GITHUB_PATH
+          else
+            echo "${CARGO_HOME:-$HOME/.cargo}/bin" >> $GITHUB_PATH
+          fi
         fi
+      env:
+        RUNNER_OS: "${{ runner.os }}"
 
     - name: rustup toolchain install ${{inputs.toolchain || 'stable'}}
       env:

